pkg/metastore/partition_collection.go

```diff
@@ -0,0 +1,59 @@
+package metastore
+
+import (
+	"fmt"
+)
+
+type PartitionCollection struct {
+	partitions []*Partition
+}
+
+func NewPartitionCollection(partitions []*Partition) *PartitionCollection {
+	return &PartitionCollection{partitions: partitions}
+}
+
+func (p *PartitionCollection) Name(i int) string {
+	return p.partitions[i].Name()
+}
+
+func (p *PartitionCollection) Len() int {
+	return len(p.partitions)
+}
+
+func (p *PartitionCollection) Less(i, j int) bool {
+	return comparePartitions(p.partitions[i], p.partitions[j]) == ItemLess
+}
+
+func (p *PartitionCollection) Swap(i, j int) {
+	p.partitions[i], p.partitions[j] = p.partitions[j], p.partitions[i]
+}
+
+func (p *PartitionCollection) Value(i int) interface{} {
+	return p.partitions[i]
+}
+
+func (p *PartitionCollection) CompareWith(i int, v interface{}, j int) CompareResult {
+	if otherIter, ok := v.(*PartitionCollection); ok {
+		return comparePartitions(p.partitions[i], otherIter.partitions[j])
+	}
+	err := fmt.Errorf("%w *PartitionCollection, got %T", ErrExpectedType, v)
+	panic(err)
+}
+
+func partitionValueEqual(partitionA, partitionB *Partition) bool {
```

Comment:
when this helper function is used inside `comparePartitions` it looks ok, but it is also used in other functions where we do not compare the partition name and I think it is missing there.